driver/SevenSegment.c

```diff
@@ -0,0 +1,218 @@
+#include "stm32xx_hal.h"
+//#include "SevenSegmentHeader.h"
```

Comment:
What was the reasoning for not including the header? I feel like you could move some of the includes and global structs to the `.h`.